app/controllers/alchemy/admin/base_controller.rb

```diff
@@ -167,6 +167,21 @@ def is_page_preview?
       def load_locked_pages
         @locked_pages = Page.from_current_site.locked_by(current_alchemy_user)
       end
+
+      # Returns the current site.
+      #
+      def current_alchemy_site
+        @current_alchemy_site ||= begin
+          site_id = params[:site_id] || session[:site_id]
+          if site_id.nil?
+            session.delete :site_id
+            Site.find_for_host(request.host)
```

Comment:
We don't need to load the site here again. The base controller already loads the site. Now, we load the site twice per admin request. 

We only need to load the site here, if we have an id in the params. And as I'm writing this, we need to think about only loading the site in the base controller if no site is is in the params, right? Otherwise we would load the sites twice if a site if param is present. We should avoid that.